zerver/tests/test_send_email.py

```diff
@@ -70,39 +74,185 @@ def test_limited_to_length(self) -> None:
+        backend = PersistentSMTPEmailBackend()
+        backend.open()
+        self.assertEqual(backend.opened_at, timezone_now())
+
+        with (
+            mock.patch.object(backend, "_open", wraps=backend._open) as _open_call,
+            mock.patch.object(backend, "close", wraps=backend.close) as close_call,
+        ):
+            # Old connection is open, but we still reopen a new connection because max connection lifetime is 0.
+            initialize_connection(backend)
+            self.assertEqual(close_call.call_count, 1)
+            self.assertEqual(_open_call.call_count, 2)
+            self.assertEqual(backend.opened_at, timezone_now())
+            close_call.reset_mock()
+            _open_call.reset_mock()
+
+            backend.close()
+            close_call.reset_mock()
+            # Old connection is closed, but we still reopen a new connection because max connection lifetime is 0.
+            initialize_connection(backend)
+            self.assertEqual(close_call.call_count, 1)
+            self.assertEqual(_open_call.call_count, 2)
+            self.assertEqual(backend.opened_at, timezone_now())
+            close_call.reset_mock()
+            _open_call.reset_mock()
+
+            backend.connection = mock.MagicMock(spec=SMTP)
+            backend.connection.noop.return_value = [250]
+            hamlet = self.example_user("hamlet")
+            with (
+                self.settings(EMAIL_HOST_USER="test", EMAIL_HOST_PASSWORD=""),
+                mock.patch.object(backend, "_send", return_value=True),
+            ):
+                send_email(
+                    "zerver/emails/password_reset",
+                    to_emails=[hamlet.email],
+                    from_name="From Name",
+                    from_address=FromAddress.NOREPLY,
+                    language="en",
+                    connection=backend,
+                )
+            # 1 call to close after sending the message because max
+            # connection lifetime is 0. 1 call to open because
+            # send_messages calls open one time.
+            self.assertEqual(close_call.call_count, 1)
+            self.assertEqual(_open_call.call_count, 1)
+
+    @time_machine.travel(timezone_now(), tick=False)
+    @override_settings(EMAIL_MAX_CONNECTION_LIFETIME_IN_MINUTES=None)
+    def test_max_connection_lifetime_none(self) -> None:
+        backend = PersistentSMTPEmailBackend()
+        backend.open()
+        self.assertEqual(backend.opened_at, timezone_now())
+
+        with (
+            mock.patch.object(backend, "_open", wraps=backend._open) as _open_call,
+            mock.patch.object(backend, "close", wraps=backend.close) as close_call,
+        ):
+            backend.connection = mock.MagicMock(spec=SMTP)
+            backend.connection.noop.return_value = [250]
+
+            # Old connection is open, we will not open a new connection because max connection lifetime is None.
```

Comment:
...but also no time has passed, so this test isn't really testing how the lifetime being None interacts with the march of time?

Reply:
Fixed!